### PR TITLE
added timeout to middleware requests

### PR DESCRIPTION
**Pull request by praffq**
## Proposed Changes

- Added a timeout setting for middleware requests with default to 20 seconds

### Associated Issue

- #2237 

## Merge Checklist

- [ ] Tests added/fixed
- [ ] Update docs in `/docs`
- [ ] Linting Complete
- [ ] Any other necessary step

_*Only PR's with test cases included and passing lint and test pipelines will be reviewed*_

@ohcnetwork/care-backend-maintainers @ohcnetwork/care-backend-admins

### CODE REVIEW

File: config/settings/base.py [hunk 1]
```diff
@@ -663,3 +663,6 @@
 TASK_SUMMARIZE_DISTRICT_PATIENT = env.bool(
     "TASK_SUMMARIZE_DISTRICT_PATIENT", default=True
 )
+
+# Timeout for middleware request (in seconds)
+MIDDLEWARE_API_TIMEOUT = 20
```
Comment (sainak) [inline config/settings/base.py]:
```suggestion
MIDDLEWARE_REQUEST_TIMEOUT = env.int("MIDDLEWARE_REQUEST_TIMEOUT", 20)
```

File: care/utils/assetintegration/base.py [hunk 2]
```diff
@@ -16,6 +16,7 @@ def __init__(self, meta):
         self.host = self.meta["local_ip_address"]
         self.middleware_hostname = self.meta["middleware_hostname"]
         self.insecure_connection = self.meta.get("insecure_connection", False)
+        self.timeout = getattr(settings, "MIDDLEWARE_API_TIMEOUT", 20)
```
Comment (sainak) [inline care/utils/assetintegration/base.py]: this is not required 
you can reference it directly from settings.MIDDLEWARE_REQUEST_TIMEOUT

**Comment by praffq** [comment]
updated

File: care/utils/assetintegration/base.py [hunk 3]
```diff
@@ -16,6 +16,7 @@ def __init__(self, meta):
         self.host = self.meta["local_ip_address"]
         self.middleware_hostname = self.meta["middleware_hostname"]
         self.insecure_connection = self.meta.get("insecure_connection", False)
+        self.timeout = settings.MIDDLEWARE_REQUEST_TIMEOUT
```
Comment (sainak) [inline care/utils/assetintegration/base.py:19]: there is no use in setting this instance var before using it, we could instead accept timeout as an optional kwarg timeout in the init method

update the children classes as well

**Review by sainak — APPROVED** [review]

**Review by vigneshhari — APPROVED** [review]

**Pull request merged**